### PR TITLE
Disabling clusters in the top level menu if the cluster is in an error state

### PR DESCRIPTION
**Pull request by codyrancher**
### Summary
Fixes rancher/dashboard#7036

### Occurred changes and/or fixed issues
This should be isolated to the top level menu where the clusters are listed

### Areas or cases that should be tested
Broken and healthy clusters
A simpler way to reproduce:
- Create a cluster with a single node
- Edit yaml of cluster
- Change `cloudCredentialSecretName: cattle-global-data:cc-4twkpo` to be a secret that doesn't exist.

### Areas which could experience regressions
This should be isolated only to the top level menu where the clusters are listed

### Screenshot/Video
image
image
image

### CODE REVIEW

**Comment by nwmac** [comment]
@codyrancher There was an issue that some dependencies got updated that broke the plugins validation action - fixes have been merged to master to address this - you'll need to rebase this PR to get the fix so that the GH Action passes

**Review by jordojordo — APPROVED** [review]
LGTM - Should we disable the ability to navigate to error'ed clusters from the mgmt and home page in a separate PR?

File: shell/components/nav/TopLevelMenu.vue [hunk 1]
```diff
@@ -182,6 +182,11 @@ export default {
     $route() {
       this.shown = false;
     },
+    shown() {
+      // We need to reload the clusters so we can get the status from the cluster relationships.
+      // Specifically this is to determine if the cluster is in an error state.
+      this.$store.dispatch('management/findAll', { type: MANAGEMENT.CLUSTER, opt: { force: true } });
```
Comment (nwmac) [inline shell/components/nav/TopLevelMenu.vue:188]: Isn't this going to make an API call every time someone clicks to show the menu?

I'm not sure we want that - don't we get the updates via the watches on the resource?

**Review by nwmac — CHANGES_REQUESTED** [review]

File: shell/components/nav/TopLevelMenu.vue (hunk 1 above)
Comment (codyrancher) [reply shell/components/nav/TopLevelMenu.vue:173]: We don't get updates from the watches. It does make a request for each click.

Comment (codyrancher) [reply shell/components/nav/TopLevelMenu.vue:173]: It looks like provisioning cluster receives updates but management cluster doesn't. I'll see if I can use provisioning cluster instead.

**Comment by codyrancher** [comment]
> LGTM - Should we disable the ability to navigate to error'ed clusters from the mgmt and home page in a separate PR?

I updated both of those places.

**Pull request merged**